### PR TITLE
upgrade requirements to resolve security alerts

### PR DESCRIPTION
**Pull request by jansenk**
**TL;DR -** This upgrades several requirements to newer versions to resolve some dependabot security alerts.

**What changed?**
upgraded jshint, karma, and karma-coverage.
that in turn upgraded http-proxy to a safe version, and removed handlebars as a requirement entirely.

**Developer Checklist**
- [x] Reviewed the [release process](./release_process.md)
- [x] Translations up to date
- [x] JS minified, SASS compiled
- [x] Test suites passing on Jenkins
- [x] Bumped version number in [setup.py](../setup.py) and [package.json](../package.json)

JIRA: [EDUCATOR-5201](https://openedx.atlassian.net/browse/EDUCATOR-5201)

FIY: @edx/masters-devs-gta

### CODE REVIEW

**Review by Andytr1 — APPROVED** [review]
Looks good.

**Review by nsprenkle — APPROVED** [review]
Looks good, tested locally
